src/internal/common/network/retry_handler.go

```diff
@@ -0,0 +1,186 @@
+package network
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"syscall"
+
+	"github.com/alcionai/clues"
+)
+
+var _ io.ReadCloser = &resetRetryHandler{}
+
+const (
+	numMaxRetries  = 3
+	rangeHeaderKey = "Range"
+	// One-sided range like this is defined as starting at the given byte and
+	// extending to the end of the item.
+	rangeHeaderOneSidedValueTmpl = "bytes=%d-"
+)
+
+// Could make this per wrapper instance if we need additional flexibility
+// between callers.
+var retryErrs = []error{
+	syscall.ECONNRESET,
```

Comment:
I don't think adding a comment about that here will really solve the issue. That's something that needs addressed on the system level. Asking every dev who may want to use this retry handler to know about all of that (even if, as you mention this component is becomes more generic and has nothing to do with networking directly) seems like a pretty high bar

I'd suggest the alternative that we start to go through and figure out a better way to handle these errors across the system as a whole rather than relying on a bunch of piecemeal solutions here and there

Reply:
Oh, I'm not imagining this as a solution to the issue.  Just a useful acknowledgement in case someone goes looking because "why is this happening?"

In fact, I'm not sure anything needs solving, per se.  The reader needs to retry if it fails.  The getter needs to retry if it fails.  The thing doing the getting may need to retry if it fails.  Those are all needs.

Then again, now that I'm coming back to this, maybe we don't need to retry on ECONNRESET in the getter.  Separation of concerns: assume that the getter implementation owns connection reset retries for the process of generating an io.ReadCloser.  Then leave this package to _only_ retry on `Read()` errors.  That way there's no conflict.

Reply:
>maybe we don't need to retry on ECONNRESET in the getter. Separation of concerns: assume that the getter implementation owns connection reset retries for the process of generating an io.ReadCloser. Then leave this package to only retry on Read() errors. That way there's no conflict.

That doesn't really seem to buy us much though since it's just special-casing something that already happens. Additionally, having the getter handle connection resets seems like it would actually worsen the problem of multiplicative retries if we're not careful (i.e. don't pass additional info to getter -> getter just naively retries with some hardcoded value). Again, I think we should be trying to consolidate retry logic like this to as few places as possible, preferably such that they're on separate code paths from each other

Reply:
On the other hand, the getter implementation cannot guarantee that some other retry client is elsewhere backing it.  I'm all for tracking centralized retires, but that sounds like scaffolding which isn't easily or cleanly threaded through the call stack in a reliable way.

Kinda feel like we're turning to bike-shedding here, so I don't want to chase this topic too far.  The likelihood of persistent econnreset errors within getter retries is extremely low.  Special case handling beyond brute force retries isn't likely to produce much value either way.